scripts/amd/gemm/README.md

```diff
@@ -1,44 +1,164 @@
+  each stage of the tuning in real-time. The default is False.
+  - `--verbose` will enable more logging message than `--time_breakdown`, such
+  as output from rocprofv2
+  - `--o OUTPUT` can be used to control the output filename to store the tuning
+  result. The default filename is `tuning_results_branchName@gitCommit_timeStamp.yaml`.
+  Therefore, each time the user runs the tuning script, a different output file
+  will be generated.
+
+Here are some example usages of running the script for tuning:
+
+Tune some gemm sizes with f16 input
+```python
+./tune_gemm --gemm_size_file input.yaml --ngpus 8 --jobs 32 --o output.yaml
 ```
+It's recommended to use as many GPUs as possible and set `--jobs` to
+a value that is 4 to 6 times the number of GPUs.
 
-2. Tune a single gemm size
-```bash
-python tune_gemm.py -m 16 -n 16 -k 16
+If you are only allowed to use a subset of the GPUs, you can
+```python
+./tune_gemm --gemm_size_file input.yaml --gpu_ids 0,1,3,4 --jobs 32 --o output.yaml
 ```
+This runs the profiling on GPU 0,1,3,4.
 
-3. Choose the file to store tuning results
-```bash
-python tune_gemm.py --gemm_size_file input_gemm_sizes.yaml --o output_tuning.yaml
+For bf8 input
+```python
+./tune_gemm --gemm_size_file input.yaml --ngpus 8 --jobs 32 -dtype_a bf8 -dtype_b bf8
 ```
 
-4. Only check correctness given the tuning results
-```bash
-python tune_gemm.py --gemm_size_file output_tuning.yaml --compare_wo_tuning
+Check correctness of the tuned configs
+```python
+./tune_gemm --gemm_size_file output.yaml --compare_wo_tuning
 ```
-Note that the tuning results file are provided as the `gemm_size_file` in this scenario.
+
+
+### Benchmark mode
+
+In benchmark mode, the script will run a single given config multiple times to
+collect performance data. The benchmark mode works as
+The tuning script works as follows
+```python
+./tune_gemm --gemm_size_file input.yaml [options] --benchmark
+```
+The supported `options` are as followings
+- `-dtype_a dtype`, `-dtype_b dtype`, and `-dtype_c dtype`: same as tuning mode.
+- `--iters n` controls the number of iterations to run the kernel.
+The default value is 1000.
+  
+
+## Tuning script implementation overview
+
+The general idea of the tuning script can be summarized as
+- Compile all the kernels in the tuning space in parallel.
+- Divide the tuning space into tasks and invoke `rocprofv2` once per
+task. This will save invocation overhead of the profiler.
+- Profile tasks in parallel on multiple GPUs.
+
+For detailed implementation, please refer to the changelog of each version.
+
+
+# Changelog
+
+## GEMM tuning script v1
+
+Shucai (@scxiao) implemented the first version of gemm tuning script: https://github.com/ROCmSoftwarePlatform/triton/pull/309
+
+## GEMM tuning script v2
+
+This version is based on v1 and @alefimov-amd's thread pool https://github.com/ROCmSoftwarePlatform/triton/pull/310
+
+### Main features
+- `rocprof` is used to measure the time for kernels in the full tuning space
```

Comment:
@xiaohuguo2023 found some advantages to using rocprofv2. Is there a reason to continue using rocprof?